src/dcd/server/autocomplete/util.d

```diff
@@ -143,8 +144,14 @@ SymbolStuff getSymbolsForCompletion(const AutocompleteRequest request,
 	ScopeSymbolPair pair = generateAutocompleteTrees(tokenArray, allocator,
 		rba, request.cursorPosition, moduleCache);
 	auto expression = getExpression(beforeTokens);
-	return SymbolStuff(getSymbolsByTokenChain(pair.scope_, expression,
-		request.cursorPosition, type), pair.symbol, pair.scope_);
+	auto symbols = getSymbolsByTokenChain(pair.scope_, expression,
+		request.cursorPosition, type);
+		if (symbols.length == 0 && doUFCSSearch(stringToken(beforeTokens.front), stringToken(beforeTokens.back))) {
+			// Let search for UFCS, since we got no hit
```

Comment:
I don't quite see why we only check for UFCS if we got no hit. If there is a worry about performance, does it actually make such a big impact? If it does, we could also think about streaming symbols instead of only sending them all out at once.

Reply:
This is actually quite important and is required to make the navigation work, if you look in tests/tc_locate_ufcs_function there is a test:
`void main()
{
	auto foo = Foo();
	foo.ufcsBar;
}`

if you remove the if statement or atleast the symbols.lenth == 0 part the test will fail.

Let say the cursor is on top if the letter u of ufcsBar

the business logic as i can see it is that the function getSymbolsForCompletion
tries to look up function of parent which is stored in "beforeTokens variable" , but since it's isn't a method of any class or struct it fails.

So it basically tries to find foo.ufcsBar, which it fails to do. we do the symbols.length == 0 to make sure that it have failed to find foo.ufscBar.

we are actually interrested in it to find ufscBar by itself.

So if there aren't any hit. so we actually make it look for the function which it actually is. 
that's how it works if it makes sense :D

Reply:
Hmm you are right I will do some changes regarding the lookup.